detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunction.kt

```diff
@@ -41,5 +43,7 @@ class UndocumentedPublicFunction(config: Config = Config.empty) : Rule(config) {
     }
 
     private fun KtNamedFunction.shouldBeDocumented() =
-        (isTopLevel || containingClassOrObject?.isPublic == true) && isPublicNotOverridden()
+        (isTopLevel || containingClassOrObject?.isPublic == true) &&
+            parents.filterIsInstance<KtClassOrObject>().none { !it.isPublic } &&
```

Comment:
```suggestion
            parents.filterIsInstance<KtClassOrObject>().all { it.isPublic } &&
```

Perhaps is more readable?